### ISSUE TITLE
[Bug] App crash randomly

### ISSUE THREAD

**Issue by bonninf** [duckduckgo/Android#1659]
### Describe the bug

Sometimes when browsing the app crash.

It seems the issue is due to the favicon files which are saved. Especially the app crash trying to update an existing favicon. When the app tries to delete the existing image file then an exception is throwed by the system but this exception is not catched by the app.

/cache/favicons/b6135a5c90439c371a28ab6f5d6a902696bd1b2524bba175f85bf07a05838a78.png: Tried to overwrite the destination, but failed to delete it.

Extract from the stack:
kotlin.io.FilesKt__UtilsKt.copyTo (Utils.kt:226)
kotlin.io.FilesKt__UtilsKt.copyTo$default (Utils.kt:217)
...

### How to Reproduce

Make a search with Duck Duck Go
Go to the a web site
Browsing during a time
Add a page as a bookmark
Some times the app crash after a while

### Expected behavior

The app should not crash

### Environment

```markdown
- DDG App Version: 5.104.0
- Device: Galaxy A20e
- OS: Android 10
```

**Comment by cmonfortep** [comment]
Does it happen for a specific site or just randomly? I use DDG as my primary browser and I haven't experienced this scenario. We would need something more specific to investigate the issue.

**Comment by bonninf** [comment]
Just randomly as it seems to me.
I didn't find a specific scenario but it happens sometimes

**Comment by swordonfire** [comment]
There is definitely a bug with Bookmarks in the Android app. But for me, it's not random.
I have a lot of bookmarks stored in it.Whenever I scroll down , the thumbnails of the sites keep refreshing .If I scroll down really slow (such that correct thumbnail is already there while I scroll past it) nothing happens...it doesn't crash. If I scroll even just a little faster that the correct thumbnails of sites are not there (that is they are still loading) then it just crashes.....I have been having this problem of app crashing while using the bookmarks menu for a while now....I was hoping that the issue would get resolved in the subsequent updates but it isn't so far. Hope it gets resolved soon.

**Comment by bonninf** [comment]
Hi @swordonfire. Thanks for your precisions. What do you mean by "a lot of ..." ? 
I tried to reproduce the crash without success. I have about 20 bookmarks knowing that my screen is able to show 6 ones at a time.

**Comment by swordonfire** [comment]
Probably more than 100...but try doing after clearing cache.The issue becomes more visible then.

**Comment by cmonfortep** [comment]
closing as fixed by #1987
